### DISCUSSION TITLE
Infinite indexing for one photo

### DISCUSSION THREAD

**Discussion by VasilyBorisovich** [photoprism/photoprism#2771] [Q&A]
infinite indexing 1 photo
Build 220901-f493607b0, install to synology nas as docker application
put one foto to /docker/photo-prism/photos
startup docker, select Library, then Start. and all. The whole night application could not index one photo
the file is normal, file manager of synology see this file, I can copy it locally. there are no errors in the logs. But this file is not visible by application.
from log (extended output) container is ok, only debug information

[root@220901-bookworm](mailto:root@220901-bookworm):/photoprism$ find ./originals/ -type f -name "*.jpg"
./originals/IMG_0014.jpg
./originals/2022/10/20221011_095600_8FCB67E4.jpg
./originals/2022/10/20221011_095600_58987973.jpg
./originals/2022/10/20221011_102236_8FCB67E4.jpg
./originals/2022/10/20221011_102236_58987973.jpg

![infinite_loop](https://user-images.githubusercontent.com/33516500/195103740-13e0ece2-8fb7-4359-b2c6-561a0008d801.png)

but last file in infinite indexing

**Comment by srett** [comment]
> from log (extended output) container is ok, only debug information

What does the ~~fox~~ log say? It would be helpful to see what exactly the last message was regarding the indexing process. Please set log level to debug and post logs.

**Reply by lastzero** [reply]
It could be that the container was restarted after an out-of-memory error. In this case, the user interface would not receive any notification that indexing is complete and, of course, the file would not be visible.

**Comment by VasilyBorisovich** [comment]
The PHOTOPRISM_DEBUG is set to true, add PHOTOPRISM_LOG_LEVEL env variable with debug value, restart container and and restarted indexing. After starting the comp is very dull, the button cancel works only with 5-6 attempts.
in 10 minutes 4 files could not be indexed. there is nothing critical in the logs. why can't I see the debug information?
in the settings it is set that the log is written to /photoprism/storage/photoprism.log but there is no such file. why?
root@220901-bookworm:/photoprism/storage$ cat /photoprism/storage/photoprism.log                                                                        
cat: /photoprism/storage/photoprism.log: No such file or directory

on the server 1GB out of 4 is free (synology nas ds918).
in the container other information
root@220901-bookworm:/photoprism/storage$ cat /proc/meminfo                                                                                             
MemTotal:        3867168 kB                                                                                                                             
MemFree:          139792 kB                                                                                                                             
MemAvailable:     172740 kB                                                                                                                             
Buffers:            6684 kB                                                                                                                             
Cached:          2365664 kB                                                                                                                             
SwapCached:        53932 kB                                                                                                                             
Active:           204120 kB                                                                                                                             
Inactive:         219604 kB    

from log tabs:
2022-10-12 09:11:01 INFO faces: loading facenet
2022-10-12 09:10:57 INFO index: found no .ppignore file
2022-10-12 09:10:49 INFO classify: loading labels from labels.txt
2022-10-12 09:08:35 WARN purge: already running (start)
2022-10-12 09:05:56 ERRO index: already running
2022-10-12 09:05:51 INFO indexing originals...
2022-10-12 09:04:40 INFO settings saved
also attached log from docker
![inf1](https://user-images.githubusercontent.com/33516500/195268367-1b420515-4cd7-402c-99a0-f397b36e774b.png)
[PhotoPrism.csv](https://github.com/photoprism/photoprism/files/9762199/PhotoPrism.csv)

**Reply by graciousgrey** [reply]
Does your server fulfill our system requirements https://docs.photoprism.app/getting-started/#system-requirements ?

**Reply by srett** [reply]
> MemAvailable: 172740 kB

That is 172MB, almost nothing. You're running out of memory.

Try adding more memory, or more swap (which could be rather slow, unfortunately, but might avoid crashes).
Also set `PHOTOPRISM_WORKERS` to `1` (even though I think PP automatically caps that value if the available RAM is too low).
Having multiple workers will double/triple/... the amount of RAM used during indexing.
You could also try setting `PHOTOPRISM_DISABLE_TENSORFLOW` to save even more RAM.

But in general, 4GB is really on the lower end of the spectrum, especially if you're running other services on the same machine.

**Reply by VasilyBorisovich** [reply]
Thanks for the quick response
while I will try on rpi4, there is 8GB of memory
